### DISCUSSION TITLE
A very simple and graphical programming language for the robots.

### DISCUSSION THREAD

**Discussion by Goldiloxx** [unitystation/unitystation#7314] [Ideas]
## Description

There could be a very simple programming language. (Something like Pure NXT (Not the computer program) or Scratch, though the latter might be kind of complicated so Pure NXT (Not the computer program) is probably the best option).

What the program will basically be is a serious of simple, predetermined* commands like go x blocks right, turn clockwise/counterclockwise, pick up an item at on of the 9 squares you're facing, attack one of the 9 squares you're facing, use a tool in your right/left hand. Have a loop and stuff like that. It would mean very complicated stuff will take a lot of work but simple yet effective programs won't be so hard to make. Will make the roboticist's job much more fun and won't require a lot of work to implement since the robots already follow preset rules, you're just changing them.

**Comment by PerfectTangent** [comment]
converting to a discussion

**Comment by Aranclanos** [comment]
Anything programmable in ss13 always ended up in tragedy and eventually removed, I discourage anyone on using time developing a feature that will be only be abused by a few utilizing the same system and ignored by the rest, with the potential of being removed once again like all other previous iterations.
